lua/orgmode/capture/init.lua

```diff
@@ -216,7 +223,14 @@ function Capture:refile_to_headline(destination_file, lines, item, headline_titl
   end
 
   if not headline then
-    return self._refile_to_end(destination_file, lines, item)
+    utils.echo_info(
+      "headline '"
+        .. headline_title
+        .. "' does not exist in "
+        .. destination_file
+        .. '. Refiling to end of file instead.'
+    )
+    return self:_refile_to_end(destination_file, lines, item)
```

Comment:
this line was the actual culprit, fyi

Reply:
For now, I would show this message only when there is a `headline_title` provided and it wasn't found. This needs some refactoring to be more readable because there's a lot of confusing abstractions.

Reply:
Added a check on `headline_title`. Still, this will never actually print, as it is overwritten by `Wrote <path/to/specified/file>`

Reply:
@kristijanhusak did some refactoring on the branching in this function. Let me know if its still unclear